web/src/server_events.js

```diff
@@ -128,10 +127,6 @@ function get_events_success(events) {
         }
     }
 
-    if (message_lists.home.selected_id() === -1 && !message_lists.home.visibly_empty()) {
-        message_lists.home.select_id(message_lists.home.first().id, {then_scroll: false});
-    }
```

Comment:
This block of code is a little puzzling -- it looks like the below from `message_list.js` is how we also select the first message when receiving a message in an empty normal message feed.

```
        if (this.narrowed && !this.visibly_empty() && this.selected_id() === -1) {                      
            // The message list was previously empty, but now isn't                                     
            // due to adding these messages, and we need to select a                                    
            // message. Regardless of whether the messages are new or                                   
            // old, we want to select a message as though we just                                       
            // entered this view.                                                                       
            const first_unread_message_id = this.first_unread_message_id();                             
            assert(first_unread_message_id !== undefined);                                              
            this.select_id(first_unread_message_id, {then_scroll: true, use_closest: true});            
        }                                                                                               
```

I'm not sure if there's something we need to do here. It would be reasonable for logic to leave `selected_id = -1` for non-visible message lists, and plan to fix that in `narrow.activate` if appropriate, but that probably requires some additional changes if it is our plan.

Reply:
Oh! I think we can just remove the `this.narrowed` condition from the `message_list.add_messages` logic and this works for the combined feed view too. Nice catch!